src/coredump/_UCD_access_reg_qnx.c FTBS for arm

**Environment**
- libunwind version: 1.8.2, master
- architecture: arm
- OS: QNX

**Description**
`src/coredump/_UCD_access_reg_qnx.c` fails to build from source for arm

**Reproduction Steps**
Steps to reproduce the behavior:
1. Get a fresh checkout
2. `./configure --host=arm-unknown-nto-qnx7.1.0eabi --build=x86_64-linux-gnu`
3. `make`
4. See error
```
  CC       coredump/_UCD_access_reg_qnx.lo
../../src/coredump/_UCD_access_reg_qnx.c: In function '_UCD_access_reg':
../../src/coredump/_UCD_access_reg_qnx.c:129:41: error: 'UNW_ARM_R16' undeclared (first use in this function); did you mean 'UNW_ARM_D16'?
   if (regnum >= UNW_ARM_R0 && regnum <= UNW_ARM_R16) {
                                         ^~~~~~~~~~~
                                         UNW_ARM_D16
../../src/coredump/_UCD_access_reg_qnx.c:129:41: note: each undeclared identifier is reported only once for each function it appears in
Makefile:4865: recipe for target 'coredump/_UCD_access_reg_qnx.lo' failed
make[2]: *** [coredump/_UCD_access_reg_qnx.lo] Error 1
```

--- a/src/coredump/_UCD_access_reg_qnx.c
+++ b/src/coredump/_UCD_access_reg_qnx.c
@@ -126,7 +126,7 @@ _UCD_access_reg (unw_addr_space_t  as,
       return -UNW_EINVAL;
   }
 #elif defined(UNW_TARGET_ARM)
-  if (regnum >= UNW_ARM_R0 && regnum <= UNW_ARM_R16) {
+  if (regnum >= UNW_ARM_R0 && regnum <= UNW_ARM_R15) {
      *valp = ui->prstatus->greg.arm.gpr[regnum];
   } else {
        Debug(0, "bad regnum:%d\n", regnum);

